galgebra/utils.py

```diff
@@ -1,18 +1,32 @@
-# utils.py
-
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
+"""
+Utility Classes 
+"""
 
+import sys
 import collections
 
+# From https://github.com/benjaminp/six/blob/master/six.py
+PY2 = sys.version_info[0] == 2
+PY3 = sys.version_info[0] == 3
+
+if PY3:
+    string_types = str
+else:
+    string_types = basestring
+
 # https://stackoverflow.com/questions/16176742/python-3-replacement-for-deprecated-compiler-ast-flatten-function
+
+
 def flatten(x):
     result = []
 
     for el in x:
-        if isinstance(x, collections.Iterable) and not isinstance(el, str):
+        if isinstance(x, collections.Iterable) and not isstr(el):
             result.extend(flatten(el))
         else:
             result.append(el)
-    return result
\ No newline at end of file
+    return result
+
+
+def isstr(s):
```

Comment:
In order to make string type test work under both python 2 and 3, I need this small util if I don't want to introduce six.

Reply:
Yeah the Python 2 to 3 string conversion is an absolute nightmare, if we get python3 support working well enough we can probably just ditch python2 support if it turns out to be an annoying problem

Reply:
So far the Python 2 support is not a big problem per se, with the help of CI and tools/libraries we can maintain good support for 2.7 & 3. Also I hope the releases from pygae won't make users having to choose between 2.7 and 3, especially we are at such an early stage of divergence.

Here's just some notes about trying to keep the dependencies minimal, so I prefered an utility function instead of introducing a dependency ( even it's a quite mature and ubiquitous one ).